### ISSUE TITLE
Should we have a section on authorised agents?

### ISSUE THREAD

**Issue by darobin** [w3ctag/privacy-principles#81]

**Comment by dmarti** [comment]
An authorized agent is a member of the same team as the user agent and shares the same duties. The authorized agent should also be a **trustworthy agent** acting in the interests of the user. Maybe this should be defined and explained in section 6, User Agents.

An authorized agent has capabilities that the user agent doesn't, because the authorized agent can operate outside the web. Sometimes the most effective agent mail is postal. 

There might be a need for some text on enabling cooperation between user agents and authorized agents? If the user has both a user agent and an authorized agent they should not get in each other's way?

 * The user might want to share a filtered version of their web history with their agent. For example, the user might want to have the browser share with the agent any domains where the user made a purchase (filled in a form with credit card info or used a web payments API) but not the full URL of the item purchased or any history that doesn't include a purchase

 * The user might want to measure the effectiveness of their authorized agent, by having their user agent cooperate in looking for signs of inappropriate cross-site personalization

**Comment by sandandsnow** [comment]
@dmarti, I think the expression I have heard most commonly is "trusted intermediaries". Out of curiosity, how would you characterize an extension such as Privacy Badger or Ghostery?

**Comment by dmarti** [comment]
@sandandsnow It seems to me that a browser extension would be part of the user agent, since it runs as if it were the user in a browser process. The browser's duty of loyalty to the user would imply that the browser does not extend itself with code that does not operate in the user's interest.

Most privacy extensions use some kind of filter list or block list. The service that hosts the list might be an example of another trusted intermediary, with the same kinds of duties to the user.

**Comment by jyasskin** [comment]
"the browser does not extend itself with code that does not operate in the user's interest" is harder than it sounds, and enforcing it gets us accused of preventing users from nominating authorised agents like adblockers. :)

I'm worried about saying that browsers should cooperate with "authorised agents" because of how easy it is to trick people into authorizing agents who don't operate in their interests.

**Comment by dmarti** [comment]
@jyasskin Yes, the ad blocker category raises a complicated set of issues. Some ad blockers claim to operate in the interests of the user, but it's possible for a user to install one without understanding the actions the ad blocker is taking. Some sites that offer Internet how-to advice often tell users who are having computer problems to "install an ad blocker" to fix a variety of issues.

It's pretty clear that that some users understand what some ad blockers do, but that is not always the case. A filter list service that includes a [paid allow-listing scheme](https://github.com/mozilla/addons/issues/171), that is technically disclosed to users but not generally well understood, is probably not a trusted intermediary, so I agree that cooperating with such a service is not generally something the browser should do.

The ad blocker code that runs in the browser would be part of the user agent, and a service that provides a filtering list for an ad blocker would be a trusted intermediary.

A CCPA authorized agent is a lot harder to authorize than a browser extension. Generally it's harder to add an authorized agent than to make an online purchase or to install an extension. You have to

* Fill out a form with your contact info
* Use codes emailed and texted to you to show that you control the phone number and email you provided
* Sign a document, either a permission letter (for CCPA opt outs) or a power of attorney (for CCPA right to know and right to delete)

If your browser and your CCPA agent are working together, you might be able to add your agent to receive a partial history sync -- but most sites are probably not going to need to be able to claim to be an agent.

**Comment by jyasskin** [comment]
I think we should fix anything in our document that would seem to say that UAs SHOULD NOT share data with an authorized agent, but I'd be reluctant to say UAs SHOULD share either, because of how hard it seems for a browser to figure out if the user has actually understood what they're authorizing an agent to do. For example, a browser on your computer isn't going to be able to tell that you read and filled out a paper form; it's going to have to trust something electronic that's vulnerable to dark patterns.

Some sort of auditing requirement could make that more plausible, but a trustworthy audit program is very complex to set up, and I think other topics deserve that cost before an "agent" program.

**Comment by dmarti** [comment]
The other problem that this touches on is that people/users/data subjects are not just individuals, but members of [families](https://pbs.twimg.com/media/EM1OOLhXYAAWEjv.jpg) and organizations, employees of firms, and have other relationships that help enforce norms. It's unrealistic to expect the data subject to go it alone while the other side has multiple parties.

Related to #2 - the owner of a device can also be a firm or the "family CIO" who acts on behalf of the user to enforce norms.

**Comment by jyasskin** [comment]
In today's meeting, we suggested that this should turn into one or more principles about how sites, authorized agents, UAs, and users should interact, but we don't yet have a solid sense of what those principles should be.

**Comment by darobin** [comment]
Put an issue (at least) about this in the _User Agents_ section.
